formats/json/commonMain/src/kotlinx/serialization/json/internal/lexer/AbstractJsonLexer.kt

```diff
@@ -246,8 +246,11 @@ internal abstract class AbstractJsonLexer {
      * Tries to consume `null` token from input.
      * Returns `true` if the next 4 chars in input are not `null`,
      * `false` otherwise and consumes it.
+     *
+     * If [moveForward] is true and if the next 4 chars in input are `null` then
+     * current position in lexer move forward on these 4 chars.
      */
-    fun tryConsumeNotNull(): Boolean {
+    fun tryConsumeNotNull(moveForward: Boolean = true): Boolean {
```

Comment:
Maybe rename `moveForward` to `consume` to align with other naming? 

Although it seems that function itself is not named properly: it says 'consumeNotNull,' but it consumes only nulls